File: 2D_amber/Assets/Scripts/JS/Behaviour/AttackBehaviour.js
#pragma strict

class AttackBehaviour extends StateMachineBehaviour{
  var transhold : float = 1 ;
  var order : int = 0 ;
  
  function OnStateUpdate(anim:Animator, info:AnimatorStateInfo, id:int){
    if(info.normalizedTime <= transhold){
      anim.SetInteger("Attack", order) ;
    }
  }
  
  function OnStateExit(anim:Animator, info:AnimatorStateInfo, id:int){
    anim.SetInteger("Attack", 0) ;
  }
}

File: 2D_amber/Assets/Scripts/CSharp/Behaviour/AttackBehaviour.cs
using UnityEngine;
using System.Collections;

public class AttackBehaviour : StateMachineBehaviour {

	public float transhold = 1.0f ;
  	public int order = 0 ;
  
  	void OnStateUpdate(Animator anim, AnimatorStateInfo info, int id){
		if(info.normalizedTime <= transhold){
			anim.SetInteger("Attack", order) ;
		}
	}
  
	void OnStateExit(Animator anim, AnimatorStateInfo info, int id){
		anim.SetInteger("Attack", 0) ;
	}
}
